bench.hpp

```diff
@@ -1,5 +1,89 @@
 #pragma once
 
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+namespace DwarfBench {
+
+/**
+ * @brief Dwarfs list enum
+ *
+ */
+enum Dwarf {
+  CONSTANT_EXAMPLE,
+  CONSTANT_EXAMPLE_C_API,
+  CONSTANT_EXAMPLE_DPCPP,
+  CUCKOO_HASH_BUILD,
+  DPL_SCAN,
+  GROUP_BY,
+  GROUP_BY_LOCAL,
+  HASH_BUILD,
+  HASH_BUILD_NON_BITMASK,
+  JOIN,
+  NESTED_LOOP_JOIN,
+  RADIX,
+  REDUCE_DPCPP,
+  TBB_SORT,
+  TWO_PASS_SCAN,
+};
+
+/**
+ * @brief Device type for execution settings
+ *
+ */
+enum DeviceType { CPU, GPU };
```

Comment:
How are we going to control the platform setup? Assume I have multiple GPUs of different vendors.

Reply:
I think that to support this we also need to change `dpcpp_common.cpp:get_device_selector` logic. We now selecting devices with default selectors from `sycl`. This enum is just an adapter between private and public API. Maybe we should create a separate issue for this and handle this in different PR?

Reply:
Sure, go for it

Reply:
Created #52